src/transformers/generation/configuration_utils.py

```diff
@@ -409,6 +409,13 @@ def validate(self, is_init=False):
                 )
 
         # 2. detect beam-only parameterization when not in beam mode
+        if self.num_beams is None:
+            warnings.warn(
+                "`num_beams` is set to None - defaulting to 1.",
+                UserWarning,
+            )
```

Comment:
We're trying to use `logger.warning` instead of `warnings` now so that users have more control over the log's verbosity

```suggestion
            logging.warning("`num_beams` is set to None - defaulting to 1.", UserWarning)
```

Reply:
Thanks, added it!